tools/c7n_azure/tests/test_keyvault.py

```diff
@@ -72,3 +74,87 @@ def test_compare_permissions(self):
         p1 = {"keys": ['get'], "secrets": ['get'], "certificates": ['get']}
         p2 = {}
         self.assertFalse(WhiteListFilter.compare_permissions(p1, p2))
+
+    @arm_template('keyvault.json')
+    @patch('c7n_azure.session.Session.get_tenant_id', return_value=DEFAULT_SUBSCRIPTION_ID)
+    def test_whitelist(self, get_tenant_id):
+        """Tests basic whitelist functionality"""
+        p = self.load_policy({
+            'name': 'test-key-vault',
+            'resource': 'azure.keyvault',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cckeyvault1*'},
+                {'not': [
+                    {'type': 'whitelist',
+                     'key': 'principalName',
+                     'users': ['account1@sample.com']}
+                ]}
+            ]
+        })
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+    @arm_template('keyvault-no-policies.json')
+    def test_whitelist_zero_access_policies(self):
+        """Tests that a keyvault with 0 access policies is processed properly
+        and doesn't raise an exception.
+        """
+        p = self.load_policy({
+            'name': 'test-key-vault',
+            'resource': 'azure.keyvault',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cckeyvault2*'},
+                {'not': [
+                    {'type': 'whitelist',
+                     'key': 'principalName',
+                     'users': ['account1@sample.com']}
+                ]}
+            ]
+        })
+        resources = p.run()
+        self.assertEqual(len(resources), 0)
+
+    @arm_template('keyvault.json')
+    @patch('c7n_azure.session.Session.get_tenant_id', return_value=DEFAULT_SUBSCRIPTION_ID)
```

Comment:
did you mean to put `DEFAULT_SUBSCRIPTION_ID` or `get_tenant_id`? these are usually different guids?

Reply:
It's purposeful in this case.

It's due to the `AzureVCRBaseTest` replacing subscription ids with `DEFAULT_SUBSCRIPTION_ID` in urls during cassette generation, that also happens to match and replace tenant ids in the url, since they're both the same pattern (guids). The underlying call being recorded here is to Microsoft Graph to resolve Azure AD details, which requires tenant id to be passed to it. 

Requests running against cassettes need to have matching urls. So `DEFAULT_SUBSCRIPTION_ID` is needed to be mocked into `get_tenant_id` for the url to be built properly at test time so that the cassette replay will work.

Also, `get_tenant_id` needed to be mocked one way or another as it was being extracted from the auth token, which doesn't exist at test time.

Make sense?

Reply:
I think to make it clear we can create `DEFAULT_TENANT_ID` constant, just to avoid confusion why `get_tenant_id` returns `subscription_id`

Reply:
Done

Reply:
We might just end up with "DEFAULT_GUID"  ;)